packages/cli-core/src/loggers/amplify_io_events_bridge.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  AmplifyIOHost,
+  AmplifyIoHostEventMessage,
+  AmplifyIoHostEventRequestMessageIoRequest,
+} from '@aws-amplify/plugin-types';
+
+/**
+ * Implements IIoHost interface of AmplifyIOHost
+ */
+export class AmplifyIOEventsBridge implements AmplifyIOHost {
+  /**
+   * a
+   */
+  constructor(
+    private readonly eventHandlers: {
+      notify?: (<T>(msg: AmplifyIoHostEventMessage<T>) => Promise<void>)[];
+      requestResponse?: (<T, U>(
+        msg: AmplifyIoHostEventRequestMessageIoRequest<T, U>
+      ) => Promise<U>)[];
+    }
+  ) {}
+
+  /**
+   * Receive amplify events and fan out notifying anyone interested.
+   */
+  notify<T>(msg: AmplifyIoHostEventMessage<T>): Promise<void> {
+    if (!this.eventHandlers.notify) {
+      return Promise.resolve();
+    }
+    const promises: Promise<void>[] = this.eventHandlers.notify?.flatMap(
+      (handler) => handler(msg)
+    );
+    return Promise.any(promises);
```

Comment:
should this be `all` ? 

i.e. why would some handler be special?

Reply:
yup, should be all. Updated